### DISCUSSION TITLE
Remove soft edge/white borders of images

### DISCUSSION THREAD

**Discussion by liveman1** [libvips/libvips#2618] [Help wanted]
Hi, 
on retina displays I get white borders in my deep zoom image using openseadragon as shown in the screenshots. They are on each zoom / pyramid level. I do not see them on screens with less resolution.

I think this is caused by tiny white or soft edges in the generated image tiles using dzsave.
As input I use pdf or png with transparent background.

`vips dzsave test.pdf[dpi=50] --layout dz --tile-size 512 --compression 0 --suffix .png test.zip`

Do you know this issue or how to do it right?
Is there any way to crop the images by 1px on the bottom & right? Overlapping does not solve it, since the white borders are just overlapped & still there.

![IMG-0382](https://user-images.githubusercontent.com/7196290/148921627-789e4ef3-e4c7-44f3-8c93-174a40912aa8.jpg)
![IMG-0380](https://user-images.githubusercontent.com/7196290/148921630-1513c363-b20f-426a-90f2-e6ac8752b4fa.jpg)
![IMG-0381](https://user-images.githubusercontent.com/7196290/148921631-8ae9c51b-b437-4e64-aa85-ca97cbeaa378.jpg)
![IMG-0379](https://user-images.githubusercontent.com/7196290/148921632-22502706-d364-49e1-9358-b5bd7f313f42.jpg)

Thanks a lot!

**Comment by jcupitt** [comment]
Hi @liveman1,

I tried with the CMU-1 demo svs:

```
$ vips dzsave CMU-1.svs x
$ vips scale x_files/16/181_129.jpeg 16.png
$ vips scale x_files/15/90_64.jpeg 15.png
```

To make:

![15](https://user-images.githubusercontent.com/580843/148823319-5ebbb476-cc3b-40b9-a264-33bc31b51bcc.png)
![16](https://user-images.githubusercontent.com/580843/148823321-583a573a-6e84-4b07-adb2-9fe7f31dd67b.png)

I don't think there are any white lines. Do you have a test image that shows this issue?

**Reply by jcupitt** [reply]
Oh heh I need stronger specs, I read SVS.

Can you post an example?

**Reply by liveman1** [reply]
sorry, I just updated my initial post.
Thanks a lot!

**Comment by jcupitt** [comment]
Ah this is a problem with overlaps. Due to rounding, you are seeing the canvas background showing through at some magnifications. There are no white lines in the tiles generated by libvips.

Make the pyramid with overlap 1 or higher, and make sure your viewer supports them.

**Reply by jcupitt** [reply]
You could try other tiles sizes too, or larger overlaps. Are you sure openseadragon is picking up the overlap setting correctly?

**Reply by liveman1** [reply]
Thanks!   There was an issue in openseadragon (https://github.com/openseadragon/openseadragon/pull/2075) which was fixed end of last year. The white lines are caused by transparent png. I will test & report, if it fixes the issue.
